src/ion/moves.rs

```diff
@@ -519,6 +502,73 @@ impl<'a, F: Function> Env<'a, F> {
                     }
                 }
 
+                // Scan debug-labels on this vreg that overlap with
+                // this range, producing a debug-info output record
+                // giving the allocation location for each label.
+                if !debug_labels.is_empty() {
+                    // Do a binary search to find the start of any
+                    // relevant labels. Recall that we require
+                    // debug-label requests to be sorted by vreg then
+                    // by inst-range.
+                    //
+                    // We search by the *end* of vale-label range as
+                    // the key in the slice (`tuple.2`), with the
+                    // *start* of this vreg range as the search key,
+                    // so we will find the first value-label range
+                    // with an end equal to or less than this range's
+                    // start. As long as we are careful to exclude the
+                    // just-adjacent case (end == start)
+                    // below, this will give us the first value-label
+                    // that possibly exists in this vreg range.
+                    let start = match debug_labels
+                        .binary_search_by_key(&(vreg.index(), range.from), |tuple| {
+                            (tuple.0.vreg(), ProgPoint::before(tuple.2))
+                        }) {
+                        // Exact match: this debug-label tuple had an
+                        // (exclusive) end exactly equal to this vreg
+                        // range's (inclusive) start; so take the next
+                        // entry as our start.
+                        Ok(i) => i + 1,
+                        // No exact match; `i` is the index at which
+                        // this key would have been inserted.
+                        Err(i) => i,
+                    };
+
+                    for &(label_vreg, label_from, label_to, label) in &debug_labels[start..] {
+                        let label_from = ProgPoint::before(label_from);
+                        let label_to = ProgPoint::before(label_to);
+                        if label_from >= range.to {
+                            break;
+                        }
+                        if label_vreg.vreg() != vreg.index() {
+                            break;
+                        }
+                        if label_from == label_to {
+                            continue;
+                        }
+                        assert!(label_from < label_to);
```

Comment:
```suggestion
                        debug_assert!(label_from < label_to);
```
I replaced all uses of `assert!` with `debug_assert!` in #25. While this particular case is unlikely to have any impact on performance, it's probably a good rule of thumb to always prefer debug-only assertions unless there is a safety invariant involved.

Reply:
Done!